### PR TITLE
Fix #756 GetValue From Lookup don't work properly

### LINKED ISSUE #756: Issue on GetValue From Lookup
I tried to run this test method from the UIAutomation.Sample project and it is failing when the lookup field contains a value. It is showing "System.InvalidOperationException: 'Field: primarycontactid Does not exist'"

public void UCITestGetValueFromLookup()
        {
            var client = new WebClient(TestSettings.Options);
            using (var xrmApp = new XrmApp(client))
            {
                xrmApp.OnlineLogin.Login(_xrmUri, _username, _password, _mfaSecrectKey);
                xrmApp.Navigation.OpenApp(UCIAppName.Sales);
                xrmApp.Navigation.OpenSubArea("Sales", "Accounts");
                xrmApp.Grid.OpenRecord(0);
                xrmApp.ThinkTime(2000);
                string lookupValue = xrmApp.Entity.GetValue(new LookupItem { Name = "primarycontactid" });
            }
        }

### PR DESCRIPTION
**Pull request by AngelRodriguez8008**
### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (updates to documentation, formatting, etc.)

### Description
Fix #756 GetValue From Lookup don't work properly
At least in the cases that I check for 5 different lookup controls.
Account -> Owner (on header)
Account -> Primary Contact 
Account -> Currency
Contact -> Account
Contact -> Company
The new  XPath to the text value in UCI look like:
`//div[data-id="[NAME].fieldControl-LookupResultsDropdown_[NAME]_selected_tag_text"]`

I found that the implementation of GetValue(LookupItem[]) from PR #515 use a similar XPath.
`{ "Entity_LookupFieldExistingValue", "//*[@data-id='[NAME].fieldControl-LookupResultsDropdown_[NAME]_selected_tag']" }`

The implementation of #515 was refactored in previos PRs, but still having the same logic, described as follow:
There is two different implementations for **GetValue** From Lookups:
1. for one value **GetValue(LookupItem)**
2. for "activity party fields", lookups that allows more than one value **GetValue(LookupItem[])**
It's look like 2 works fine for also 1. But the current implementation of 1. is no longer working (#756)

Fixed on the on the top of the previous PR #754 one [commit](https://github.com/microsoft/EasyRepro/commit/8b8d47478eed2eb7d77a5be195b0e2a163f88e59), 2 files.

This PR Unify the Logic of GetValue with LookupItem & LookupItem[]
The new Implementation, use in both cases the same logic but:
**GetValue(LookupItem)** 
returns for a Single Value Lookup an String Value or an Empty String (as before)
but also returns a "; " separated values for a Multi-Values-Lookup

**GetValue(LookupItem[])** 
returns for a Single Value Lookup an String[] with just one Value or an Empty String[]
returns for a Multi-Values-Lookup an String[] (as before)

Optional: this logic can also be adapted to return "null" instead empties string & string[].

### Issues addressed
Fix #754

### All submissions:

- [x] My code follows the code style of this project.
- [x] Do existing samples that are effected by this change still run?
- [x] I have added samples for new functionality. 
- [x] I raise detailed error messages when possible.
- [x] My code does not rely on labels that have the option to be hidden.

### Which browsers was this tested on?

- [x] Chrome
- [ ] Firefox
- [ ] IE
- [ ] Edge

### CODE REVIEW

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs [hunk 1]
```diff
@@ -392,7 +392,7 @@ public static class AppElements
             { "Entity_Header_FlyoutButton","//button[contains(@id,'headerFieldsExpandButton')]" },
             { "Entity_Header_LookupFieldContainer", "//div[@data-id='header_[NAME].fieldControl-Lookup_[NAME]']" },
             { "Entity_Header_TextFieldContainer", "//div[@data-id='header_[NAME].fieldControl-text-box-container']" },
-            { "Entity_Header_OptionSetFieldContainer", "//div[@data-id='header_[NAME].fieldControl-option-set-container']" },
+            { "Entity_Header_OptionSetFieldContainer", "//div[@data-id='header_[NAME]']" },
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs:395]: Fix #755

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs [hunk 2]
```diff
@@ -329,7 +329,7 @@ public static class AppElements
             { "Entity_CloseOpportunityWin"       , "//button[contains(@data-id,'MarkAsWon')]"},
             { "Entity_CloseOpportunityLoss"       , "//button[contains(@data-id,'MarkAsLost')]"},
             { "Entity_Delete"       , "//button[contains(@data-id,'Delete')]"},
-            { "Entity_FormContainer"       , "//*[@id=\"tab-section\"]"},
+            { "Entity_FormContainer"       , "//*[@data-id='editFormRoot']"},
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs:332]: Used for check if the form is loaded

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs [hunk 3]
```diff
@@ -277,12 +277,12 @@ public static class AppElements
...
             { "Nav_SiteMapLauncherButton", "//button[@data-lp-id=\"sitemap-launcher\"]" },
-            { "Nav_SiteMapLauncherCloseButton", "//button[@aria-label=\"Close Site Map\"]" },
+            { "Nav_SiteMapLauncherCloseButton", "//button[@data-id='navbutton']" },
             { "Nav_SiteMapAreaMoreButton", "//button[@data-lp-id=\"sitemap-areaBar-more-btn\"]" },
             { "Nav_SiteMapSingleArea", "//li[translate(@data-text,'ABCDEFGHIJKLMNOPQRSTUVWXYZ','abcdefghijklmnopqrstuvwxyz') = '[NAME]']" },
             { "Nav_AppMenuContainer"       , "//*[@id=\"taskpane-scroll-container\"]"},
-            { "Nav_SettingsLauncherBar"       , "//*[@data-id='[NAME]Launcher']"},
-            { "Nav_SettingsLauncher"       , "//*[@data-id='[NAME]']"},
+            { "Nav_SettingsLauncherBar"       , "//button[@data-id='[NAME]Launcher']"},
+            { "Nav_SettingsLauncher"       , "//div[@id='[NAME]Launcher']"},
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs:285]: Fix Navigation Tests

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs [hunk 4]
```diff
@@ -277,12 +277,12 @@ public static class AppElements
             { "Nav_WebAppMenuButton"       , "//*[@id=\"TabArrowDivider\"]/a"},
             { "Nav_UCIAppMenuButton"       , "//button[@data-id=\"navbar-switch-app\"]"},
             { "Nav_SiteMapLauncherButton", "//button[@data-lp-id=\"sitemap-launcher\"]" },
-            { "Nav_SiteMapLauncherCloseButton", "//button[@aria-label=\"Close Site Map\"]" },
+            { "Nav_SiteMapLauncherCloseButton", "//button[@data-id='navbutton']" },
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/AppElementReference.cs:280]: Look like "Close Site Map" button was replaced, not able to find in page.

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/Elements/Entity.cs [hunk 5]
```diff
@@ -523,9 +523,9 @@ public void SelectForm(string formName)
         /// </summary>
         /// <param name="controls">The activityparty lookup field name, value or index of the lookup.</param>
         /// <example>xrmApp.Entity.AddValues(new LookupItem[] { new LookupItem { Name = "to", Value = "A. Datum Corporation (sample)" } });</example>
-        public void AddValues(LookupItem[] controls, int index = 0)
+        public void AddValues(LookupItem[] controls)
         {
-            _client.AddValues(controls, index);
+            _client.AddValues(controls);
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/Elements/Entity.cs:528]: Having a extra parameter index make confusing what index to use. I keep just the index inside the control

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 6]
```diff
@@ -458,14 +458,14 @@ internal BrowserCommandResult<bool> OpenSubArea(string area, string subarea, int
...
-                var success = TryOpenSubArea(subarea);
+                var success = TryOpenSubArea(driver, subarea);
                 if (!success)
                 {
                     success = TryOpenArea(area);
                     if (!success)
                         throw new InvalidOperationException($"Area with the name '{area}' not found. ");
 
-                    success = TryOpenSubArea(subarea);
+                    success = TryOpenSubArea(driver, subarea);
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:468]: TryOpenSubArea is a private auxiliar methods, is not needed to create a Command for that, pass the driver.

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 7]
```diff
@@ -486,21 +485,25 @@ public BrowserCommandResult<bool> OpenSubArea(string subarea)
...
             var found = navSubAreas.TryGetValue(subarea, out var element);
             if (found)
-                element.Click(true);
-
+            {
+                var strSelected = element.GetAttribute("aria-selected");
+                bool.TryParse(strSelected, out var selected);
+                if (!selected)
+                    element.Click(true);
+            }
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:506]: Check if the Area is already expanded

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 8]
```diff
@@ -517,18 +520,22 @@ public BrowserCommandResult<bool> OpenArea(string subarea)
...
+            bool found = areas.TryGetValue(area, out menuItem);
+            if (found)
+            {
+                var strSelected = menuItem.GetAttribute("aria-checked");
+                bool selected;
+                bool.TryParse(strSelected, out selected);
+                if (!selected)
+                    menuItem.Click(true);
+            }
+            return found;
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:535]: Check if the Area is already expanded

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 9]
```diff
@@ -594,13 +601,7 @@ private bool TryOpenArea(string area)
...
                 {
                     var singleItem = driver.FindElement(By.XPath(AppElements.Xpath[AppReference.Navigation.SiteMapSingleArea].Replace("[NAME]", area)));
-                    char[] trimCharacters =
-                    {
-                        '', '\r', '\n', '',
-                        '', ''
-                    };
-
-                    dictionary.Add(singleItem.Text.Trim(trimCharacters).ToLowerString(), singleItem);
+                    dictionary.Add(singleItem.Text.ToLowerString(), singleItem);
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:604]: Trim Logic moved to ToLowerString

File: Microsoft.Dynamics365.UIAutomation.Browser/BrowserOptions.cs [hunk 10]
```diff
@@ -152,7 +152,8 @@ public virtual EdgeOptions ToEdge()
         {
             var options = new EdgeOptions()
             {
-                PageLoadStrategy = PageLoadStrategy.Normal
+                PageLoadStrategy = PageLoadStrategy.Normal,
+                UseInPrivateBrowsing = PrivateMode
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Browser/BrowserOptions.cs:156]: see Issue #733 this  [commet](https://github.com/microsoft/EasyRepro/issues/733#issuecomment-587234144)

File: Microsoft.Dynamics365.UIAutomation.Browser/Extensions/SeleniumExtensions.cs [hunk 11]
```diff
@@ -570,6 +571,26 @@ public static IWebElement WaitUntilClickable(this ISearchContext driver, By by,
...
+
+            bool success = driver.WaitUntil(condition);
+            if (success)
+                successCallback?.Invoke(elements);
+            else
+                failureCallback?.Invoke();
+
+            return elements;
+        }
+
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Browser/Extensions/SeleniumExtensions.cs:593]: Add WaitUntil to find Collections of Elements

File: Microsoft.Dynamics365.UIAutomation.Browser/Extensions/StringExtensions.cs [hunk 12]
```diff
@@ -39,8 +39,18 @@ public static string ToUnsecureString(this SecureString secureString)
...
+                '\n', 
+                (char) 60644, // 
+                (char) 60932, // 
+                (char) 59540, // 
+                (char) 60038, // 
+                (char) 61424, // 
+                (char) 59902, //
+            };
+
+            return value?.Trim()
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Browser/Extensions/StringExtensions.cs:53]: More readable character values

**Review by AngelRodriguez8008 — COMMENTED** [review]
Trying to Fix Alls samples Tests: Fix Some Bugs, take advantage of the new Logic in Selenium Extensions methods. Refactor some methods & remove some duplicated code.  Fix some Navigation Issues.

**Review by TYLEROL — APPROVED** [review]

**Comment by thathi** [comment]
@AngelRodriguez8008 

                    string[] lookupValues = existingValues.Select(v => v.GetAttribute("innerText").ToLowerString()).ToArray(); //IE can return line breaks

Convert tolower is it intentional in         private string[] TryGetValue(IWebElement fieldContainer, LookupItem[] controls) method?

**Pull request merged**